src/App.tsx

```diff
@@ -1,26 +1,377 @@
+  // const [shouldFocusInput, setShouldFocusInput] = useState(true);
+
+  const newTodoFieldRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    const fetchInitialData = async () => {
+      try {
+        const todosResponse = await getTodos();
+
+        setTodos(todosResponse);
+      } catch (errorTodo) {
+        setError('Unable to load todos');
+        setTimeout(() => {
+          setError(null);
+        }, 3000);
+      }
+    };
+
+    fetchInitialData();
+  }, []);
+
+  useEffect(() => {
+    if (newTodoFieldRef.current) {
+      newTodoFieldRef.current.focus();
+    }
+  }, [todos.length, loadingTodoId, tempTodo]);
+
+  const filteredTodos = useMemo(() => {
+    switch (filter) {
+      case 'Active':
+        return todos.filter(todo => !todo.completed);
+      case 'Completed':
+        return todos.filter(todo => todo.completed);
+      case 'All':
+      default:
+        return todos;
+    }
+  }, [todos, filter]);
+
+  const todosCounter = useMemo(() => {
+    const incompleteTodosCount = todos.filter(todo => !todo.completed).length;
+
+    return `${incompleteTodosCount} ${incompleteTodosCount === 1 ? 'item' : 'items'} left`;
+  }, [todos]);
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const trimmedText = inputText.trim();
+
+    if (!trimmedText) {
+      setError('Title should not be empty');
+      setTimeout(() => {
+        setError(null);
+      }, 3000);
+      // setShouldFocusInput(true);
+
+      return;
+    }
+
+    setTempTodo({
+      id: 0,
+      title: trimmedText,
+      completed: false,
+      userId: USER_ID,
+    });
+
+    try {
+      const newTodo = await postTodos(trimmedText, USER_ID);
+
+      setTodos(prevTodos => [...prevTodos, newTodo]);
+      setTempTodo(null);
+      setError(null);
+      setInputText('');
+      // setShouldFocusInput(true);
+    } catch (errorMessage) {
+      setError('Unable to add a todo');
+      setTimeout(() => {
+        setError(null);
+      }, 3000);
+      setTempTodo(null);
+      // setShouldFocusInput(true);
+    }
+  };
+
+  const handleDelete = async (id: number) => {
+    setLoadingTodoId(id);
+    try {
+      await deleteTodos(id);
+      setTodos(prevTodos => prevTodos.filter(todo => todo.id !== id));
+      setError(null);
+    } catch (errorTodo) {
+      setError('Unable to delete a todo');
+      setTimeout(() => {
+        setError(null);
+      }, 3000);
+    } finally {
+      setLoadingTodoId(null);
+    }
+  };
+
+  const handleClearCompleted = async () => {
+    const completedTodos = todos.filter(todo => todo.completed);
+    const deletePromises = completedTodos.map(todo =>
+      deleteTodos(todo.id).then(
+        () => ({ status: 'fulfilled', id: todo.id }),
+        errorTodo => ({ status: 'rejected', id: todo.id, errorTodo }),
+      ),
+    );
+
+    const results = await Promise.all(deletePromises);
```

Comment:
I guess you can use `Promise.allSettled` instead - https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled